lib/SILOptimizer/Mandatory/PerformanceDiagnostics.cpp

```diff
@@ -689,16 +764,25 @@ class PerformanceDiagnosticsPass : public SILModuleTransform {
       if (function.wasDeserializedCanonical())
         continue;
 
-      // Don't check generic functions in embedded Swift, they're about to be
-      // removed anyway.
-      if (getModule()->getOptions().EmbeddedSwift &&
-          function.getLoweredFunctionType()->getSubstGenericSignature())
-        continue;
-
       if (function.getPerfConstraints() == PerformanceConstraints::None) {
         diagnoser.checkNonAnnotatedFunction(&function);
       }
     }
+
+    if (getModule()->getOptions().EmbeddedSwift) {
+      for (SILFunction &function : *module) {
+        // Don't check constructors and destructors directly, they will be
+        // checked if called from other functions, with better source loc info.
```

Comment:
Not all constructors and destructors are directly called. For example, class destructors are implicitly called from `destroy_value` instructions.
Excluding them here means that they are potentially not checked at all.

Reply:
Hm, good point. What if instead of not checking them, I only changed the *order* in which we go over the SILFunctions, so that it's something like (1) public/external functions first, (2) vtable members without constructors/destructors, (3) constructors/destructors?

The motivation is strictly to get better diagnostics with better "called from here" information, I don't want to skip any checks.

Reply:
>  I only changed the order in which we go over the SILFunctions

Sounds good!